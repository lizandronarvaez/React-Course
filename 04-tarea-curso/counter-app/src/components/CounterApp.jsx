/* eslint-disable no-unused-vars */
import React, { useState } from 'react'
import PropTypes from 'prop-types'
const CounterApp = ({ value }) => {
    // useState
    const [counter, setCounter] = useState(value)
    // handleOperations
    const handleAdd = () => setCounter(counter + 1);
    const handleSubtract = () => setCounter(counter - 1);
    const handleReset = () => setCounter(value)
    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button
                onClick={handleAdd} >
                Sumar +1
            </button>
            <button
                onClick={handleSubtract}>
                Restar -1
            </button>
            <button onClick={handleReset}>
                Reestablecer
            </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp