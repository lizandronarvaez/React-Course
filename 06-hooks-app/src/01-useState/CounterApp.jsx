import { useState } from "react"

const CounterApp = () => {

    const [counters, setCounter] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30
    })

    const { counter1, counter2, counter3 } = counters;

    return (
        <>
            <h1>Contador nº1: {counter1} </h1>
            <h1>Contador nº2: {counter2} </h1>
            <h1>Contador nº3: {counter3} </h1>
            <button
                className="btn btn-primary btn-lg"
                onClick={() => setCounter({
                    ...counters,
                    counter1: counter1 + 1
                })}
            >
               Incrementar +1
            </button>
        </>
    )
}

export default CounterApp