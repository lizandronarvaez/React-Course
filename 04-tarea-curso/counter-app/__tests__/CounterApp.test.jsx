/* eslint-disable no-undef */
import { fireEvent, render, screen } from "@testing-library/react";
import CounterApp from "../src/components/CounterApp"

describe('Pruebas en el componente <CounterApp />', () => {
    const DefaultValue = 100;

    // Test para comprobar que hace un snapshot o imagen de el archivo
    test('Debe hacer un snapshot', () => {
        const { container } = render(<CounterApp value={DefaultValue} />)
        expect(container).toMatchSnapshot();
    });

    // Test que comprueba si el valor que devuelve es el inicial
    test('Debe retornar el valor inicial', () => {

        render(<CounterApp value={DefaultValue} />);

        expect(screen.getByRole("heading", { level: 2 }).innerHTML).toBe(DefaultValue.toString())
        // console.log(screen.getByRole("heading",{level:2}).innerHTML)
        // screen.debug();
    });

    // test para comprobar que el valor aumenta cuando se hace click en el boton
    test('Al hacer click en el boton incemento, debe incrementar el valor', () => {

        render(<CounterApp value={DefaultValue} />);

        fireEvent.click(screen.getByText("Sumar +1"));

        expect(screen.getByText("101")).toBeTruthy();
    });

    // Test para comprobar que hace el decremento
    test('Al hacer click en el boton incemento, debe incrementar el valor', () => {

        render(<CounterApp value={DefaultValue} />);

        fireEvent.click(screen.getByText("Restar -1"));
        // screen.debug()
        expect(screen.getByText("99")).toBeTruthy();
    });

    test('Al hacer click en reestablecer debe volver a colocarse el valor inicial', () => {

        render(<CounterApp value={DefaultValue} />);

        fireEvent.click(screen.getByText("Sumar +1"));
        fireEvent.click(screen.getByText("Reestablecer"));
        fireEvent.click(screen.getByText("Restar -1"));
        fireEvent.click(screen.getByText("Sumar +1"));
        fireEvent.click(screen.getByRole("button", {
            name: "btn-resetgit"
        }))

        // screen.debug()
        expect(screen.getAllByText(DefaultValue)).toBeTruthy();

    });
});