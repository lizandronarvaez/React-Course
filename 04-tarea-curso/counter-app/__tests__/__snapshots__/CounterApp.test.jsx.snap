// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CounterApp /> Debe hacer un snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button>
    Sumar +1
  </button>
  <button>
    Restar -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reestablecer
  </button>
</div>
`;
